File: LeetCode/Arrays and Strings/Missing Number/JavaScript Solution/MissingNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    const naturalSum = (nums.length * (nums.length + 1)) / 2; let sum = 0;
    for(let num of nums) {
        sum += num;
    }
    
    return naturalSum - sum;
};

File: LeetCode/Arrays and Strings/Missing Number/C++ Solution/missing_number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int sum1 = 0, sum2 = 0;
        sum1 = (nums.size() * (nums.size() + 1)) / 2;
        
        for(int i = 0 ; i < nums.size(); i++) {
            sum2 += nums[i];
        }
        
        return sum1 - sum2;
    }
};